android/tests_backend/widgets/base.py

```diff
@@ -0,0 +1,40 @@
+from pytest import skip
+
+
+class SimpleProbe:
+    def __init__(self, widget, container):
+        for i in range(container._impl.native.getChildCount()):
+            child = container._impl.native.getChildAt(i)
+            if child is widget._impl.native:
+                self.native = child
+                assert isinstance(self.native, self.native_class)
+                break
+        else:
+            raise ValueError(f"cannot find {widget} in {container}")
```

Comment:
The idea behind "simple" was that it covers the case where there's a 1-1 relationship between Toga widgets and native widgets. But I know you've mentioned that some Toga widgets on some platforms might be implemented as multiple native widgets. Are there any widgets like this already? If so, their probes couldn't inherit from this base class, because there wouldn't just be a single `self.native` property to implement all the methods against.

Reply:
> However, this test is a specific test of the type of layout it's been added to.

It's no longer making any assumptions except that the widget is somewhere inside the container. What kind of layout would that not work with?

The idea behind this code is to verify that adding a Toga widget to a Toga container causes a native widget of the correct type to be added to the native container. Or in the case of a Toga widget which produced multiple native widgets, the counterpart to this code would also need to verify that *all* the native widgets had been added to the layout, and in the correct configuration.

> Pulling this block of code out as an `assert_has_child(container, widget)` assertion in the main test app (and invoked in the probe setup) would also mean we're being explicit about what has been asserted.

Good point, and that would also allow us to call the code multiple times on the same widget, to verify that a widget can be moved from one container to another. So I'll move it out of the constructor.

Reply:
> Are there any widgets like this already? If so, their probes couldn't inherit from this base class, because there wouldn't just be a single `_impl.native` property to implement all the methods against.

There are several widgets that do this. On macOS, the scrolling behavior for MultilineText is independent of the widget itself, so the "native" is the scroller, and there's a secondary widget that is the text view. Switch is another example; this time, on macOS it's a single widget, but on GTK it's a Label and a Switch, wrapped up in a Box.